calisphere/record.py

```diff
@@ -0,0 +1,217 @@
+
+    if format =='image':
+        content_file.update(get_iiif(media['media_key']))
+    if format == 'file':
+        content_file.update({'id': f"thumbnails/{thumbnail_md5}"})
+    if format in ['audio', 'video']:
+        content_file.update({
+            'id': f"thumbnails/{thumbnail_md5}",
+            'url': f"{settings.UCLDC_NUXEO_THUMBS}media/{media['media_key']}"
+        })
+    return content_file
+
+
+class Record(object):
+    def __init__(self, indexed_record, child_index=None, index='es'):
+        self.index = index
+        self.indexed_record = indexed_record
+        self.display = deepcopy(indexed_record)
+
+        if 'reference_image_dimensions' in self.indexed_record:
+            split_ref = self.display['reference_image_dimensions'].split(':')
+            self.display['reference_image_dimensions'] = split_ref
+
+        self.display['harvest_type'] = 'hosted' if self.is_hosted() else 'harvested'
+
+        if self.is_complex():
+            formats = set([
+                c['format'] for c in self.get_children() if 'format' in c])
+            all_images = all(f == 'image' for f in formats)
+
+            complex_object_display = {
+                'selected': not bool(child_index),
+                'structMap': self.get_children(),
+                'componentCount': len(self.get_children()),
+                'multiFormat': True if len(formats) > 1 else False,
+                'hasComponentCaptions': False if all_images else True,
+                'has_fixed_thumb': bool(
+                    self.indexed_record.get('reference_image_md5')),
+            }
+            if child_index:
+                complex_object_display['selectedComponentIndex'] = child_index
+                complex_object_display['selectedComponent'] = self.get_child_metadata(
+                    int(child_index))
+
+            self.display.update(complex_object_display)
+
+        if self.is_harvested():
+            oac_display = {'oac': False}
+
+            url_item = self.indexed_record.get('url_item', '')
+            if url_item.startswith('http://ark.cdlib.org/ark:'):
+                oac_display = {
+                    'oac': True,
+                    'url_item': url_item.replace(
+                        'http://ark.cdlib.org/ark:',
+                        'http://oac.cdlib.org/ark:'
+                    ) + '/?brand=oac4'
+                }
+            self.display.update(oac_display)
+
+        self.collections = [
+            Collection(col_id, index) 
+            for col_id in self.indexed_record.get('collection_ids')
+        ]
+        self.repositories = [
+            Repository(repo_id, index) 
+            for repo_id in self.indexed_record.get('repository_ids')
+        ]
+
+    def is_hosted(self):
+        if self.index == 'solr':
+            return bool(self.indexed_record.get('structmap_url'))
+        elif self.index == 'es':
+            return bool(self.indexed_record.get('media'))
```

Comment:
```suggestion
            return bool(self.indexed_record.get('mapper_type') == 'nuxeo.nuxeo')
```

Reply:
I made this change here:
https://github.com/ucldc/public_interface/pull/295/files#diff-070924927e1cadaac0c17b98fb7f54f4997438cdac87c01c696bfcac3f2b6e80R152